ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java

```diff
@@ -67,17 +70,26 @@ private static KsqlConfig loadServerConfig(final File configFile) {
     return new KsqlConfig(serverProps);
   }
 
-  public static List<Pair<byte[], byte[]>> loadBackup(final File file) throws IOException {
+  public static List<Pair<byte[], byte[]>> loadBackup(
+      final File file,
+      final RestoreOptions options
+  ) throws IOException {
     final BackupReplayFile commandTopicBackupFile = BackupReplayFile.readOnly(file);
+    List<Pair<byte[], byte[]>> records = commandTopicBackupFile.readRecords();
 
-    final List<Pair<byte[], byte[]>> records = commandTopicBackupFile.readRecords();
-    throwOnInvalidRecords(records);
+    if (options.isSkipIncompatibleCommands()) {
+      records = removeIncompatibleCommands(records);
+    }
```

Comment:
Ok I changed it so that we always check that the commands are valid. The only different with the skip option is that if it's set to true, instead of throwing an exception and exiting the program when failing to deserialize the command due to a SerializationException/IncomaptibleKsqlCommandVersionException (which indicates an incompatible command) the method continues by skipping the command.